content/08-mongodb/16-mongodb-database-tools.mdx

```diff
@@ -0,0 +1,172 @@
+
+`mongostat` returns many fields fields, and it can be customized to return only the fields of interest. Another important option to know is `--rowcount=<number>, -n=<number>`. This option limits the amount of rows returned by `mongostat`. Some examples of the fields returned are:
+
+- `inserts` : The number of objects inserted into the database per second. 
+- `query` : The number of query operations per second.
+- `vsize` : The amount of virtual memory in megabytes used by the process at the time of the last `mongostat` call. 
+- `repl` : The replication status of the member.
+
+There are many more fields covered in the [Official MongoDB Documentation](https://www.mongodb.com/docs/database-tools/mongostat/#fields), but these few examples demonstrate the `mongostat` utility’s capability for database monitoring from the system command-line.
+
+### mongotop
+While `mongostat` is a useful tool for monitoring on a database level, `mongotop` is a useful tool to know for providing statistics on a per-collection level. Specifically, `mongotop` provides a method to track the amount of time a `mongod` instance spends reading and writing data every second.
+
+`mongotop` can only be run from the command line, and its basic syntax looks as follows:
+
+	mongotop <options> <connection-string> <polling interval in seconds>
+
+`mongotop` returns the following fields:
+
+- `mongotop.ns` : This is the database namespace, which is a combination of the database name and collection
+- `mongotop.total` : Provides the total amount of time that the `mongod` spent operation on the namespace.
+- `mongotop.read` : Provides the amount of time that the `mongod` spent performing read operation on the namespace.
+- `mongotop.write` : Provides the amount of time that the `mongod` spent performing write operation on the namespace.
+- `mongotop.<timestamp>` : Provides the time stamp for the returned data.
+
+`mongotop` allows a database user to monitor the traffic of a collection within a database. You’ll be able to form an image of when the collection is experiencing spikes or lulls in read or write operations.
+
+## GridFS
+[GridFS](https://www.mongodb.com/docs/manual/core/gridfs/) is a convention for storing large files in a MongoDB database. All of the official MongoDB drivers support this convention, as does the following `mongofiles` program. It acts as an abstraction layer for storage and recovery of large files such as videos, audios, and images. 
+
+### mongofiles
+The `mongofiles` tool makes it possible to manipulate files stored in a MongoDB instance in GridFS objects from the system’s command line. This is particularly useful because it provides an interface between the objects stored in your file system and GridFS.
```

Comment:
```suggestion
The `mongofiles` tool makes it possible to manipulate files stored in a MongoDB instance as GridFS objects from the system’s command line. This is particularly useful because it provides an interface between the objects stored in your file system and GridFS.
```

I was having trouble parsing the multiple clauses here but I honestly have no idea if this is correct semantically.